File: LeetCode_3120/Main.py
class Solution(object):
    def numberOfSpecialChars(self, word):
        """
        :type word: str
        :rtype: int
        """
        lower_set = set()
        upper_set = set()
        
        for char in word:
            if char.islower():
                lower_set.add(char)
            elif char.isupper():
                upper_set.add(char.lower())
        
        count = 0
        for char in lower_set:
            if char in upper_set:
                count += 1
        
        return count

File: LeetCode_3120/Main.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int numberOfSpecialChars(string word) {
        unordered_set<char> lowerSet;
        unordered_set<char> upperSet;
        
        for (char c : word) {
            if (islower(c)) {
                lowerSet.insert(c);
            } else if (isupper(c)) {
                upperSet.insert(tolower(c));
            }
        }
        
        int count = 0;
        for (char c : lowerSet) {
            if (upperSet.count(c)) count++;
        }
        
        return count;
    }
};
